services/yaml.py

```diff
@@ -25,6 +25,15 @@ def fetch_commit_yaml(commit: Commit, owner: Owner | None) -> Dict | None:
     Fetches the codecov.yaml file for a particular commit from the service provider.
     Service provider API request is made on behalf of the given `owner`.
     """
+
+    # There's been a lot of instances where this function is called where the owner arg is not Owner type
+    # Previously this issue is masked by the catch all exception handling. Most badly typed calls have
+    # been addressed, but there might still be some entrypoints unaccounted for, will fix new discoveries
+    # from this assertion being raised.
+    assert owner is None or isinstance(
+        owner, Owner
+    ), f"fetch_commit_yaml owner arg must be Owner or None. Provided: {type(owner)}"
```

Comment:
Although we have a lot of `asserts` in the code, using this for checks is generally not good practice (i.e. can be turned off in production for performance). Can we explicitly raise a `TypeError` here instead?

Reply:
makes sense to me. done.